.github/workflows/pythonpackage.yml

```diff
@@ -37,17 +37,19 @@ jobs:
         coveralls
     - name: flake8
       run: |
-         flake8 dedupe tests
-    - name: Static Testing
+         flake8 dedupe tests benchmarks/benchmarks
+    - name: mypy
       run: |        
         mypy --ignore-missing-imports dedupe
-    - name: Integration test, record link
+    - name: Benchmarking & Integration tests
+      # From https://github.com/pandas-dev/pandas/blob/60ac973e64ba364b345aea22bf988272536aa42f/.github/workflows/asv-bot.yml
       run: |
-        python tests/canonical_gazetteer.py -vv
-        python tests/canonical_matching.py -vv
-    - name: Integration test, dedupe
-      run: |
-        python tests/canonical.py -vv
+        cd benchmarks
+        asv check -E existing
+        git remote add upstream https://github.com/dedupeio/dedupe.git
```

Comment:
what is happening here and on line 50?

Reply:
in line 52, ASV checks out `upstream/master`, installs dedupe, and runs the benchmarks. then checks out `HEAD`, installs dedupe, runs the benchmarks. So lines 49 and 50 ensure that the ref of `upstream/master` exist. Possible to accomplish the same thing a different way probably, but this hardcodes it instead of relying on a variable, the "base commit of the PR". Honestly I was just copying from the pandas example I linked in a comment above

Reply:
so, in this configuration we are just comparing a PR to master. we aren't keeping a longer term record of performance?

Reply:
Yes. See longer explanation in general PR discussion section